presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayEqualOperator.java

```diff
@@ -43,6 +43,9 @@ public static Boolean equals(
             @SqlType("array(E)") Block leftArray,
             @SqlType("array(E)") Block rightArray)
     {
+        if (rightArray == null) {
```

Comment:
Why are the null checks needed after the upgrade?

Reply:
Fastutil now supports null keys. So all of the methods does the following check.

if (strategy.equals(key, null)){
// handle the null case
} else {
//handle the non null case
}

Reply:
Could you please elaborate?

`@SqlType("array(E)") Block rightArray ` is not marked as `@SqlNullable`, from what I remember they should never be null.

Reply:
FastUtil's lets the user specify custom equals (Strategy) and custom hashcode methods for the collection. 

public void add(Key k) {
    if (strategy.equals(key, null)) {
    } else {
    }
}

So all the strategy implementation needs to handle null, though the application may choose to not store nulls.

I don't understand why fastutil is using the strategy to do the null check, instead of using the == operator for null check. I am not sure if anyone really wants to customize the null check. I opened this issue in the fastutil's codebase
https://github.com/vigna/fastutil/issues/241